Creating a purchase order the product description doesn't respect partner language

agb80 opened the pull request:
Mirroring PR odoo/odoo#6892

Commit:
To do the quotations the product description was written but not according with the language of the user

Styles

pedrobaeza commented on addons/purchase/purchase.py:
```diff
@@ -1018,7 +1017,11 @@ def onchange_product_id(self, cr, uid, ids, pricelist_id, product_id, qty, uom_i
             dummy, name = product_product.name_get(cr, uid, product_id, context=context_partner)[0]
             if product.description_purchase:
-                name += '\n' + product.description_purchase
+                desciption_purchase_label = ir_translation._get_source(
```
Why don't you use lang key on context to get the correct translation?

agb80 replied:
`context` is holding the lang for current user, but the intention is to use the partner language, and `lang` variable is used to populate `context_partner`

pedrobaeza replied:
Create another context then. You can re-browse with the context you want.

agb80 replied:
Sorry @pedrobaeza I don't get what you mean, if you look at [line 1014](https://github.com/OCA/OCB/pull/238/files#diff-fe8587929089e5fdb6c75687011b1c7eR1014) the product object is already browsed using context with partner lang key and is not getting the correct translation, that's why we invoke the actual translation here, something similar to what is being doing [here](https://github.com/OCA/OCB/blob/7.0/addons/account/account_invoice.py#L670)

pedrobaeza replied:
OK, I get it. Please correct the variable name to `description_purchase_label` or even you can just call it `description`

agb80 commented:
@pedrobaeza  sorry for late answer, correction done!

StefanRijnhart commented:
I cannot reproduce the problem. Using OCB revision 78964636b

StefanRijnhart commented:
I see on the upstream PR that someone else came to the same conclusion. :-1: for now then.

ManulReyes committed:
fix(purchase): Fix taken translation

Take the correct translation for product in description field.This it will take the description of
description_purchase field

StefanRijnhart commented:
Closing after a year without a clear description how to reproduce.
